devtools/zenodo/zenodo_data_release.py

```diff
@@ -259,11 +266,27 @@ def sync_directory(self, source_dir: SourceData) -> "ContentComplete":
         # the files before getting to this state, so EmptyDraft would become
         # InProgressDraft.
 
-        for name, blob in source_dir.iter_files():
-            response = self.zenodo_client.create_bucket_file(
-                bucket_url=bucket_url, file_name=name, file_content=blob
+        files = fsspec.open_files(globbed, mode="rb")
+        if len(files) == 0:
+            raise ValueError(
+                f"No files found in {globbed}! Did you pass the right dirname?"
             )
-            logger.info(f"Upload to {bucket_url}/{name} got {response.text}")
+        for openable_file in files:
+            name = Path(openable_file.path).name
+            with openable_file.open() as of:
+                if "local" not in openable_file.fs.protocol:
```

Comment:
Should clean this up a bit. `fsspec` *tries* to paper over the differences between remote/local, but sometimes they're actually important (like when we don't want to do a remote read() on the file every time we need to retry due to Zenodo).

So, I should extract this into some sort of "get the file that we want to upload - which will either be the local file itself, or a tempfile we got from a remote place."